src/sign.rs

```diff
@@ -0,0 +1,127 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+pub use ed25519::*;
+
+pub mod ed25519 {
+    use std::{
+        cmp::Ordering,
+        convert::Infallible,
+        fmt::{self, Debug},
+        hash::{Hash, Hasher},
+    };
+
+    use sodiumoxide::utils;
+
+    #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+    pub struct PublicKey(pub [u8; 32]);
+
+    impl AsRef<[u8]> for PublicKey {
+        fn as_ref(&self) -> &[u8] {
+            &self.0
+        }
+    }
+
+    #[derive(Clone, Copy)]
+    pub struct Signature(pub [u8; 64]);
+
+    impl PartialEq for Signature {
```

Comment:
I’m curious. Why don’t we derive these?

Reply:
The derive macros are only defined up to `std::array::LengthAtMost32` :lobster: :shrug: :rage1: :sob: